tests/stub/notifications_config/test_session_notifications_config.py

```diff
@@ -58,6 +58,34 @@ def test_session_overrides_driver_config_on_read_tx(self):
                                    script, self.driver_params)
             self._server.reset()
 
+    def test_session_config_with_database_not_specified(self):
+        uris = [self._uri, self._uri.replace("bolt", "neo4j")]
+        for uri in uris:
+            with self.subTest(uri=uri):
+                try:
+                    host = self._server.address
+                    script_path = self.script_path(
+                        "notifications_config_session_router_run.script"
+                    )
+                    script_vars = {
+                        "#NOTIS#": '"notifications_minimum_severity"'
+                                   ': "WARNING", ',
+                        "#HOST#": host
+                    }
+                    self._server.start(script_path, vars_=script_vars)
+                    driver = Driver(self._backend, uri, self._auth)
+                    session = driver.session(
+                        None,
+                        notifications_min_severity="WARNING"
+                    )
+                    cursor = session.run("RETURN 1 as n")
+                    cursor.consume()
+                    self._server.done()
+                finally:
+                    self._server.reset()
+                    if driver is not None:
```

Comment:
Note that when the script doesn't make it to line 76
```python
driver = Driver(self._backend, uri, self._auth)
```
`driver` will not be `None` but undefined => Exception. You'd have to define `driver = None` before the try to be 100% this works as intended.